### PR TITLE
MAGECLOUD-3357: [Cloud Docker] Changes in .magento.app.yaml should affect php modules	

### PR DESCRIPTION
**Pull request by arhiopterecs**
https://magento2.atlassian.net/browse/MAGECLOUD-3357
### Description
Adds support for configuring PHP modules locally via .magento.app.yaml

### Fixed Issues (if relevant)
1. magento/magento-cloud-docker#MAGECLOUD-3357: [Cloud Docker] Changes in .magento.app.yaml should affect php modules	

### Related Pull Request
https://github.com/magento/ece-tools/pull/541

### Manual testing scenarios
https://jira.corp.magento.com/browse/MAGECLOUD-126

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Command/Generate/Php.php [hunk 1]
```diff
@@ -84,47 +117,88 @@ public function execute(InputInterface $input, OutputInterface $output)
...
+                    $dockerPhpExtEnable[] = $phpExtName;
+                }
+            }
         }
 
-        $dockerfile = $destination . '/Dockerfile';
-        $content = strtr(
+        $packges = array_merge(
+            self::EDITION_CLI == $edition ? self::DEFAULT_PACKAGES_PHP_CLI : self::DEFAULT_PACKAGES_PHP_FPM
+            , $extOsDependencies
```
Comment (shiftedreality) [inline src/Command/Generate/Php.php]: coma should be on prev line

File: data/php-extensions.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            Php::EXTENSION_OS_DEPENDENCIES => ['libxslt1-dev'],
+            Php::EXTENSION_TYPE => Php::EXTENSION_TYPE_CORE,
+        ]
+    ],
+//        'yaml' => [],
+//        'zbarcode' => [],
+//        'zendopcache' => [],
+    'zip' => ['7.*' => [Php::EXTENSION_TYPE => Php::EXTENSION_TYPE_CORE]],
+    'pcntl' => ['*' => [Php::EXTENSION_TYPE => Php::EXTENSION_TYPE_CORE]],
+];
```
[the remark this answers is not in the document]
Comment (arhiopterecs) [reply data/php-extensions.php]: Done

File: src/Command/Generate/Php.php [hunk 3]
```diff
@@ -84,47 +117,88 @@ public function execute(InputInterface $input, OutputInterface $output)
...
-
-        if (in_array($version, ['7.0', '7.1'])) {
-            $extensions = array_merge(
-                $extensions,
-                ['mcrypt', 'pcntl']
-            );
+        $this->filesystem->put($dockerfile, $this->buildDockerfile($dockerfile, $version, $edition));
+    }
+
+    private function buildDockerfile(string $dockerfile, string $phpVersion, string $edition): string
```
[the remark this answers is not in the document]
Comment (arhiopterecs) [reply src/Command/Generate/Php.php]: Done

File: src/Command/Generate/Php.php [hunk 4]
```diff
@@ -84,47 +117,88 @@ public function execute(InputInterface $input, OutputInterface $output)
...
+                if ($phpConstraintObject->matches($phpExtConstraintObject)) {
+                    switch ($phpExtInstallConfig[self::EXTENSION_TYPE]) {
+                        case self::EXTENSION_TYPE_CORE:
+                            $dockerPhpExtInstall[] = $phpExtInstallConfig[self::EXTENSION_PACKAGE_NAME] ?? $phpExtName;
+                            break;
+                        case self::EXTENSION_TYPE_PECL:
+                            $phpPeclExtensions[] = $phpExtInstallConfig[self::EXTENSION_PACKAGE_NAME] ?? $phpExtName;
+                            break;
+                        default:
+                            throw new \RuntimeException("PHP Extension type not supported");
```
[the remark this answers is not in the document]
Comment (arhiopterecs) [reply src/Command/Generate/Php.php]: Done

File: src/Command/Generate/Php.php (hunk 1 above)
Comment (arhiopterecs) [reply src/Command/Generate/Php.php]: Done

File: src/Command/Generate/Php.php [hunk 5]
```diff
@@ -84,47 +117,96 @@ public function execute(InputInterface $input, OutputInterface $output)
...
+                if ($phpConstraintObject->matches($phpExtConstraintObject)) {
+                    switch ($phpExtInstallConfig[self::EXTENSION_TYPE]) {
+                        case self::EXTENSION_TYPE_CORE:
+                            $dockerPhpExtInstall[] = $phpExtInstallConfig[self::EXTENSION_PACKAGE_NAME] ?? $phpExtName;
+                            break;
+                        case self::EXTENSION_TYPE_PECL:
+                            $phpPeclExtensions[] = $phpExtInstallConfig[self::EXTENSION_PACKAGE_NAME] ?? $phpExtName;
+                            break;
+                        default:
+                            throw new \RuntimeException('PHP Extension type not supported');
```
Comment (oshmyheliuk) [inline src/Command/Generate/Php.php]: Which extension and which type? Please add more information to the exception message.

Comment (arhiopterecs) [reply src/Command/Generate/Php.php]: Done

File: data/php-extensions.php [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+            Php::EXTENSION_OS_DEPENDENCIES => ['libyaml-dev'],
+        ],
+    ],
+    'zip' => [
+        '>=7.0.0 <7.3.0' => [Php::EXTENSION_TYPE => Php::EXTENSION_TYPE_CORE],
+    ],
+    'pcntl' => [
+        '>=7.0.0 <7.3.0' => [Php::EXTENSION_TYPE => Php::EXTENSION_TYPE_CORE],
+    ],
+//    'amqp' => [],
```
Comment (NadiyaS) [inline data/php-extensions.php]: why did you leave all these comments?

File: data/php-extensions.php [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+//    'pdo_odbc' => [],
+//    'pdo_sqlsrv' => [],
+//    'pinba' => [],
+//    'pthreads' => [],
+//    'snmp' => [],
+//    'sourceguardian' => [],
+//    'spplus' => [],
+//    'sqlsrv' => [],
+//    'sybase' => [],
+//    'tideways' => [],
```
[the remark this answers is not in the document]
Comment (arhiopterecs) [reply data/php-extensions.php]: modules will be deleted

File: data/php-extensions.php (hunk 6 above)
Comment (arhiopterecs) [reply data/php-extensions.php]: comments will be deleted

**Pull request merged**